### PR TITLE
feat: bootstrapper more resilient with retries

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [ ] My change requires a documentation update and I have done it
- [x] I have added tests to cover my changes.

### Screenshots (if appropriate):

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2997)

### CODE REVIEW

File: pkg/node/bootstrap.go [hunk 1]
```diff
@@ -199,22 +204,46 @@ func bootstrapNode(
...
 
-	if len(eventsJSON) != int(l) {
+	for i := 0; i < getSnapshotRetries; i++ {
+		reader, l, err = joiner.New(ctx, ns, snapshotReference)
+		if err != nil {
+			logger.Errorf("bootstrap: fetching snapshot: %v", err)
+			continue
+		}
+
+		eventsJSON, err = ioutil.ReadAll(reader)
```
Comment (mrekucci) [inline pkg/node/bootstrap.go:233]: The `ioutil` package is deprecated use `io` package instead.

File: pkg/postage/batchstore/reserve.go [hunk 2]
```diff
@@ -105,18 +105,19 @@ func (s *store) cleanup() error {
 
 	for _, b := range evictions {
 
-		s.logger.Debugf("batchstore: cleaning up batch %x", b.ID)
-
 		err := s.evictFn(b.ID)
 		if err != nil {
+			s.logger.Errorf("batchstore: cleaning up batch %x: %v", b.ID, err)
```
Comment (acud) [inline pkg/postage/batchstore/reserve.go:110]: don't log and return an error. instrument with `fmt.Errorf` instead

Comment (acud) [reply pkg/postage/batchstore/reserve.go:110]: same below

File: pkg/node/bootstrap.go [hunk 3]
```diff
@@ -199,22 +207,47 @@ func bootstrapNode(
...
+
+		eventsJSON, err = io.ReadAll(reader)
+		if err != nil {
+			logger.Errorf("bootstrap: reading: %v", err)
+			continue
+		}
+
+		if len(eventsJSON) != int(l) {
+			err = errDataMismatch
+			logger.Errorf("bootstrap: %v", err)
```
[the remark this answers is not in the document]
Comment (istae) [reply pkg/node/bootstrap.go:244]: The goal was to log info on why a retry was needed. I'd agree with you if the return was immediate.

File: pkg/node/bootstrap.go [hunk 4]
```diff
@@ -199,22 +207,47 @@ func bootstrapNode(
...
+		l                 int64
+		eventsJSON        []byte
+	)
 
-	reader, l, err := joiner.New(ctx, ns, snapshotReference)
-	if err != nil {
-		return nil, err
+	for i := 0; i < getSnapshotRetries; i++ {
+		snapshotReference, err = getLatestSnapshot(ctx, ns, snapshotFeed)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (acud) [reply pkg/node/bootstrap.go:219]: not addressed. you also used a similar pattern below (except the break statement is still inside an else clause). might be good to have things with similar style (at least on the same PR)

**Review by acud — APPROVED** [review]

**Pull request merged**